tests/degradation_test_v2.py

```diff
@@ -0,0 +1,175 @@
+            N = len(x)
+            minutes = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/(365 * 24 * 60), minutes)
+        elif (input_freq == 'S'):
+            x = pd.date_range(start='2012-01-01 00:00:00', end='2012-01-01 00:10:00', freq=input_freq)
+            N = len(x)
+            seconds = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/(365 * 24 * 60 * 60), seconds)
+        else :
+            sys.exit("Unknown frequency type")
+        
+        corr_energy = pd.Series(data=y, index=x)
+        return corr_energy
+
+    def get_CD_YOY_corr_energy(self, rd, input_freq):
+        ''' Create input for degradation_classical_decomposition and degradation_year_on_year
+            functions, depending on frequency. Allowed frequencies for both of these degradation 
+            functions  are 'MS', 'M', 'W' and 'D'.
+        '''
+        if (input_freq == 'MS'):
+            x = pd.date_range(start='2012-01-01', end='2015-01-01', freq=input_freq)
+            N = len(x)
+            months = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/12, months)
+        elif (input_freq == 'M'):
+            x = pd.date_range(start='2012-01-31', end='2015-01-31', freq=input_freq)
+            N = len(x)
+            months = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/12, months)
+        elif (input_freq == 'W'):
+            x = pd.date_range(start='2012-01-01', end='2015-01-01', freq=input_freq)
+            N = len(x)
+            weeks = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/52, weeks)
+        elif (input_freq == 'D'):
+            x = pd.date_range(start='2012-01-01', end='2015-01-01', freq=input_freq)
+            N = len(x)
+            days = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/365, days)
+        else :
+            sys.exit("Unknown frequency type")
+        
+        corr_energy = pd.Series(data=y, index=x)
+        return corr_energy
+
+
+    def setUp(self):
+        # define module constants and parameters
+
+        # Allowed frequencies for degradation_ols
+        self.list_ols_input_freq = ['MS','M','W','D','H','T','S']
+        # Allowed frequencies for degradation_classical_decomposition and degradation_year_on_year
+        self.list_CD_YOY_input_freq = ['MS','M','W','D']
+
+        self.rd = -0.005
+     
+        test_ols_corr_energy = {}
+        test_CD_YOY_corr_energy = {}
+        
+        for input_freq in self.list_ols_input_freq:
+            corr_energy = self.get_ols_corr_energy(self.rd, input_freq)
+            test_ols_corr_energy[input_freq] = corr_energy
+
+        self.test_ols_corr_energy = test_ols_corr_energy
+        
+        for input_freq in self.list_CD_YOY_input_freq:
+            corr_energy = self.get_CD_YOY_corr_energy(self.rd, input_freq)
+            test_CD_YOY_corr_energy[input_freq] = corr_energy
+         
+        self.test_CD_YOY_corr_energy = test_CD_YOY_corr_energy
+
+    def tearDown(self):
```

Comment:
We can remove the tearDown since we are not using it. This function is useful for cleaning up variables after each test is run.